### PR TITLE
added basic 3mf export

### LINKED ISSUE #452: Add 3MF export to manifoldCAD.org
Let's try to use @hrgdavor's https://www.npmjs.com/package/@jscadui/3mf-export to allow output suitable for 3D-printing. We can probably ignore all the color/material property stuff for now - that's probably better in GLB anyway.

### PR DESCRIPTION
**Pull request by pca006132**
Closes #452.

The 3mf export is now run together with glb export, it seems that it doesn't take much time (0.2s for gyroid) so I did not bother to make it async.

### CODE REVIEW

**Comment by hrgdavor** [comment]
I am jealous of @elalish and how good manifold has become and how quickly. Also thanks @pca006132 for taking this over the finish line :)

File: bindings/wasm/examples/worker.ts
Comment (elalish) [inline bindings/wasm/examples/worker.ts:478]: I think this is the mimetype: [`application/vnd.ms-package.3dmanufacturing-3dmodel+xml`](http://www.iana.org/assignments/media-types/application/vnd.ms-package.3dmanufacturing-3dmodel+xml)

**Review by elalish — CHANGES_REQUESTED** [review]
Thanks!

File: bindings/wasm/examples/worker.ts
Comment (pca006132) [reply bindings/wasm/examples/worker.ts:463]: I was thinking about potentially changing the message type later and did not bother to fix the type here. Will fix.

File: bindings/wasm/examples/worker.ts
Comment (pca006132) [reply bindings/wasm/examples/worker.ts:472]: I think we can, but I am not very familiar with this. We probably need to look into the 3mf spec about how to specify the hierarchy and materials.

File: bindings/wasm/examples/index.html
Comment (pca006132) [reply bindings/wasm/examples/index.html:52]: I can do this

File: bindings/wasm/examples/worker.ts
Comment (pca006132) [reply bindings/wasm/examples/worker.ts:472]: I think this may get complicated. What about leaving this to a follow up PR?

File: bindings/wasm/examples/index.html
Comment (pca006132) [reply bindings/wasm/examples/index.html:52]: Oh it seems that the dropdown you are using is very manual. I thought that it would be some sort of a reusable component similar to bootstrap.

Not sure what is the cleanest way of implementing this, I guess I will leave it for another PR.

File: bindings/wasm/examples/worker.ts
Comment (elalish) [reply bindings/wasm/examples/worker.ts:472]: Yeah, actually it looks like we also need a small change in `to3dmodel` - @hrgdavor where can we file issues and/or make PRs to `3mf-export`? Looking at your code, it appears a component tree can't be exported correctly because it places every component in the build, rather than just the root.

Comment (hrgdavor) [reply bindings/wasm/examples/worker.ts:472]: code is in this repo. Suggestions and PR's are welcome https://github.com/hrgdavor/jscadui/
3mf export part is here https://github.com/hrgdavor/jscadui/tree/main/file-format/3mf-export

File: bindings/wasm/examples/worker.ts
Comment (elalish) [inline bindings/wasm/examples/worker.ts:477]: I don't think there's any way around this for `files`, but let's prefer e.g. `results.threeMFURL`.

**Review by elalish — APPROVED** [review]
I definitely want to tweak the UX and get the node hierarchy exporting, but probably best not to block this one. I haven't had a chance to test it, so please give it a good prodding before merging.

File: bindings/wasm/examples/worker.ts
Comment (pca006132) [reply bindings/wasm/examples/worker.ts]: sure, I forgot to change this previously

**Pull request merged**
